artemis-server/src/main/java/org/apache/activemq/artemis/core/server/MessageReference.java

```diff
@@ -38,6 +38,10 @@ public static MessageReference createReference(Message encode, final Queue queue
 
    Message getMessage();
 
+   default long getMessageID() {
```

Comment:
-1 on hot path default methods on hot path aren't so good, suggest either leaving the code as was in ServerConsumerImpl with the getMessage().getMessageId or simply not make it default, and update other impl's within the code base.

nudge @franz1981

Reply:
Thanks, will change it later

Reply:
I agree here...

Reply:
I have removed default method.